lib/msf/core/module_manager/cache.rb

```diff
@@ -81,14 +81,15 @@ def load_cached_module(type, reference_name)
     if module_info
       parent_path = module_info[:parent_path]
 
+      # XXX borked
```

Comment:
I should have asked on earlier code review, but what's borked in this function in its final form? Or was this referring to the fix below where it's not possible for the `loadable?` method on a module loader to actually figure things are truly loadable in advance, since the external loader has to also execute the module to really know?

Reply:
IIRC, it's because this is somehow dependent on the order the module loaders are checked. If the external module loader is checked first then things implicitly done by the classic directory loader don't happen, like payload module generation. My initial intent was to run the external module loader first to avoid needing a reliable "is this a script check" in the classic module loader, but that didn't pan out.